extensions/mysql/lib/mysql.go

```diff
@@ -0,0 +1,439 @@
+package lib
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"github.com/go-sql-driver/mysql"
+	"github.com/pkg/errors"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/internal/tracelog"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path"
+	"path/filepath"
+	"sort"
+	"strings"
+	"time"
+)
+
+type Uploader struct {
+	*internal.Uploader
+}
+type Backup struct {
+	*internal.Backup
+}
+
+type ArchiveNonExistenceError internal.ArchiveNonExistenceError
+
+func HandleStreamFetch(backupName string, folder internal.StorageFolder) {
+	if backupName == "" || backupName == "LATEST" {
+		latest, err := internal.GetLatestBackupName(folder)
+		if err != nil {
+			tracelog.ErrorLogger.Fatalf("Unable to get latest backup %+v\n", err)
+		}
+		backupName = latest
+	}
+	stat, _ := os.Stdout.Stat()
+	if (stat.Mode() & os.ModeCharDevice) == 0 {
+	} else {
+		tracelog.ErrorLogger.Fatalf("stout is a terminal")
+	}
+	err := downloadAndDecompressStream(folder, backupName)
+	if err != nil {
+		tracelog.ErrorLogger.Fatalf("%+v\n", err)
+	}
+}
+
+// TODO : unit tests
+func downloadAndDecompressStream(folder internal.StorageFolder, fileName string) error {
+	baseBackupFolder := folder.GetSubFolder(internal.BaseBackupPath)
+	backup := Backup{internal.NewBackup(baseBackupFolder, fileName)}
+
+	streamSentinel, err := backup.fetchStreamSentinel()
+	if err != nil {
+		return err
+	}
+	binlogsAreDone := make(chan error)
```

Comment:
It doesn't show us, whether binlogs are done or not. It shows us, whether we had an error during binlogs fetching process.